test/rockpaperscissors.js

```diff
@@ -0,0 +1,717 @@
+
+    const getBlockTimeStamp = async(txObj) => {
+
+        const tx = await getTransaction(txObj.tx);
+        const blockData = await web3.eth.getBlock(tx.blockNumber)
+
+        return blockData.timestamp;
+    }
+
+    const checkEventNotEmitted = async () => {
+        const result = await truffleAssert.createTransactionResult(rcp, rcp.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result
+        );
+    };
+
+    const getGasCost = async txObj => {
+        const tx = await getTransaction(txObj.tx);
+
+        return toBN(txObj.receipt.gasUsed).mul(toBN(tx.gasPrice));
+    };
+
+    const [contractOwner, playerOne, playerTwo, playerThree] = accounts;
+    const playerOneSecretBytes32 = await soliditySha3(playerOneSecretString);
+    const playerTwoSecretBytes32 = await soliditySha3(playerTwoSecretString);
+    const wrongSecretBytes32 = await soliditySha3(wrongSecretString);
+
+    let rcp;
+    let gameKey;
+    let playerTwoMoveHash;
+    let snapshotId;
+
+    beforeEach("Deploy and prepare", async function() {
+        rcp = await Game.new({from: contractOwner});
+        gameKey = await rcp.createPlayerOneMoveHash(playerOneSecretBytes32, ROCK, {from: playerOne});
+        playerTwoMoveHash = await rcp.createPlayerTwoMoveHash(gameKey, playerTwoSecretBytes32, PAPER, {from: playerTwo});
+        const snapShot = await timeMachine.takeSnapshot();
+        snapshotId = snapShot['result'];
+    });
+
+    afterEach(async() => {
+        await timeMachine.revertToSnapshot(snapshotId);
+    });
+
+    it("createPlayerOneMoveHash() hashes the player's address, secret and move", async () => {
+        const expectedMoveHash =  await soliditySha3(playerOne, playerOneSecretBytes32, 0);
+        assert.strictEqual(expectedMoveHash, gameKey);
+    });
+
+    it("Player one creates a game", async () => {
+        const txObj = await rcp.movePlayerOne(gameKey, toBN(10), {from: playerOne, value: toBN(10)});
+
+        await truffleAssert.eventEmitted(txObj, 'PlayerOneMoves', (ev) => {
+
+            return  ev.player === playerOne &&
+                    ev.gameKey === gameKey &&
+                    ev.stake.toString(10) === "10" &&
+                    ev.amount.toString(10) === "10"
+        },'PlayerOneMoves event is emitted');
+
+        const game = await rcp.games(gameKey);
+
+        assert.strictEqual(game.stake.toString(10), "10");
+        assert.strictEqual(game.playerOne, playerOne);
+        assert.strictEqual(game.step.toString(10), PLAYER_ONE_MOVE);
+
+        const playerOneOwed = toBN(await rcp.balances(playerOne));
+
+        assert.strictEqual(playerOneOwed.toString(10), "0");
+    });
+
+    it("Value sent that exceeds the stake, is sent to the player's balance", async () => {
+        await rcp.movePlayerOne(gameKey, toBN(5), {from: playerOne, value: toBN(7)});
+
+        const playerOneOwed = await rcp.balances(playerOne);
+        assert.strictEqual(playerOneOwed.toString(10), "2");
```

Comment:
Make sure with a test that if there already is some owed, then the new piece gets added, not overwritten.